backend/src/routes/AllowlistEntry.ts

```diff
@@ -17,21 +20,46 @@ function AllowlistEntryRoutes(allowlistService: typeof AllowlistEntry) {
 
   router.post('/', requireAdmin, async (req, res) => {
     try {
-      const { phoneNumber, destinationCountry } = req.body;
+      let validatedReq;
+      try {
+        const bodySchema = z.object({
+          phoneNumber: z.string(),
+          destinationCountry: z.string(),
+        });
+        const body = bodySchema.parse(req.body);
+        validatedReq = { body };
```

Comment:
zod removes undefined keys by default so `validatedReq` will be exactly as defined with typing.